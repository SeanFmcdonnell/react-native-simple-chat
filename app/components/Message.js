import React from 'react';
import { StyleSheet, Text } from 'react-native';

class Message extends React.Component {

    constructor(props) {
        super(props);
    }

    render() {
        return (
            <Text
                style={ [styles.message, this.props.align === 'left' ? styles.messageLeft : styles.messageRight] }>
                { this.props.text.trim() }
            </Text>
        );
    }
}

const styles = StyleSheet.create({
    message: {
        textAlign: 'left',
        borderWidth: 0,
        borderRadius: 12,
        padding: 8,
        overflow: 'hidden',
        marginBottom: 5,
        maxWidth: 200
    },
    messageLeft: {
        backgroundColor: '#CDCFD0'
    },
    messageRight: {
        backgroundColor: 'rgb(0, 122, 255)',
        color: 'white'
    }
});

export {
    Message
};
