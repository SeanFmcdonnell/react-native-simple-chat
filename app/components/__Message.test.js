import React from 'react';
import { Message } from './Message';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer.create(<Message text={ 'hello' }/>).toJSON();
    expect(tree).toMatchSnapshot();
});
