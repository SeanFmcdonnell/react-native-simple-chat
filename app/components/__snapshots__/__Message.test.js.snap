// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "borderRadius": 12,
        "borderWidth": 0,
        "marginBottom": 5,
        "maxWidth": 200,
        "overflow": "hidden",
        "padding": 8,
        "textAlign": "left",
      },
      Object {
        "backgroundColor": "rgb(0, 122, 255)",
        "color": "white",
      },
    ]
  }
>
  hello
</Text>
`;
